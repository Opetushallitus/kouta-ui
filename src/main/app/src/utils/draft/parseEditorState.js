import { convertFromHTML } from 'draft-convert';
import { EditorState } from 'draft-js';
import { isString } from 'lodash';
import linkDecorator from './linkDecorator';
import createEmptyEditorState from './createEmptyEditorState';

const createEditorStateWithContent = contentState =>
  EditorState.createWithContent(contentState, linkDecorator);

const parseEditorstate = value => {
  if (!isString(value)) {
    return createEmptyEditorState();
  }

  return createEditorStateWithContent(convertFromHTML(value));
};

export default parseEditorstate;
