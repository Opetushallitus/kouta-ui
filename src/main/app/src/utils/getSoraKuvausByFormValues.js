import get from 'lodash/get';
import pick from 'lodash/pick';
import mapValues from 'lodash/mapValues';

import serializeEditorState from './draft/serializeEditorState';

const getKielivalinta = values => get(values, 'kieliversiot') || [];

const getSoraKuvausByFormValues = values => {
  const { tila, muokkaaja } = values;
  const kielivalinta = getKielivalinta(values);
  const nimi = pick(get(values, 'tiedot.nimi') || {}, kielivalinta);
  const kuvaus = pick(get(values, 'tiedot.kuvaus') || {}, kielivalinta);
  const koulutustyyppi = get(values, 'koulutustyyppi') || null;
  const julkinen = Boolean(get(values, 'julkisuus'));

  return {
    tila,
    muokkaaja,
    nimi,
    julkinen,
    koulutustyyppi,
    kielivalinta,
    metadata: {
      kuvaus: mapValues(kuvaus, serializeEditorState),
    },
  };
};

export default getSoraKuvausByFormValues;
