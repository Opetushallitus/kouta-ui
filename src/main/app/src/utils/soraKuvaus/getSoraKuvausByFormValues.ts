import _fp from 'lodash/fp';

import {
  pickTranslations,
  pickAndSerializeTranslations,
} from '../pickTranslations';

const getSoraKuvausByFormValues = values => {
  const { tila, muokkaaja, koulutusala, koulutukset } = values;

  const kielivalinta = values?.kieliversiot ?? [];

  return {
    organisaatioOid: values?.organisaatioOid?.value,
    externalId: _fp.isEmpty(values?.externalId) ? null : values?.externalId,
    tila,
    muokkaaja,
    nimi: pickTranslations(values?.tiedot?.nimi, kielivalinta),
    koulutustyyppi: values?.koulutustyyppi || null,
    kielivalinta,
    metadata: {
      koulutusalaKoodiUri: koulutusala?.value,
      koulutusKoodiUrit: koulutukset?.map(_fp.prop('value')),
      kuvaus: pickAndSerializeTranslations(
        values?.tiedot?.kuvaus,
        kielivalinta
      ),
    },
  };
};

export default getSoraKuvausByFormValues;
