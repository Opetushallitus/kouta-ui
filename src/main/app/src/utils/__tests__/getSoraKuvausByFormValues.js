import getSoraKuvausByFormValues from '../getSoraKuvausByFormValues';
import parseEditorState from '../draft/parseEditorState';

test('getSoraKuvausByFormValues returns correct sora-kuvaus given form values', () => {
  const soraKuvaus = getSoraKuvausByFormValues({
    tila: 'tallennettu',
    muokkaaja: '1.1.1.1',
    kieliversiot: ['fi', 'sv'],
    tiedot: {
      nimi: {
        fi: 'Fi nimi',
        sv: 'Sv nimi',
      },
      kuvaus: {
        fi: parseEditorState('<b>Kuvaus fi</b>'),
        sv: parseEditorState('<i>Kuvaus sv</i>'),
      },
    },
    julkinen: true,
    koulutustyyppi: 'amm',
  });

  expect(soraKuvaus).toMatchSnapshot();
});
