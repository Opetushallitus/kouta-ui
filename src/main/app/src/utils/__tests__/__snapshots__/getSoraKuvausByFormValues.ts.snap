// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getSoraKuvausByFormValues returns correct sora-kuvaus given form values 1`] = `
Object {
  "julkinen": true,
  "kielivalinta": Array [
    "fi",
    "sv",
  ],
  "koulutustyyppi": "amm",
  "metadata": Object {
    "kuvaus": Object {
      "fi": "<p><strong>Kuvaus fi</strong></p>",
      "sv": "<p><em>Kuvaus sv</em></p>",
    },
  },
  "muokkaaja": "1.1.1.1",
  "nimi": Object {
    "fi": "Fi nimi",
    "sv": "Sv nimi",
  },
  "tila": "tallennettu",
}
`;
